crd-convert/src/main/java/io/strimzi/kafka/crd/convert/converter/VersionConversion.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.crd.convert.converter;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.strimzi.api.annotations.ApiVersion;
+
+public class VersionConversion<T extends HasMetadata> extends ListConversion<T, Conversion<T>> {
+
+    private final ReplaceApiVersion<HasMetadata> replaceApiVersion;
+
+    VersionConversion(ApiVersion from, ApiVersion to) {
+        this(from, to, Collections.emptyList());
+    }
+
+    VersionConversion(ApiVersion from, ApiVersion to, List<Conversion<T>> conversions) {
+        super(conversions);
+        Objects.requireNonNull(from);
+        Objects.requireNonNull(to);
+        Objects.requireNonNull(conversions);
+        if (from.compareTo(to) >= 0) {
+            //throw new IllegalArgumentException("to must be > from");
```

Comment:
why it's commented?

Reply:
@tombentley ^^ ?

Reply:
I can't remember that. Try uncommenting it and seeing if any of the unit tests break. If not then probably keep it uncommented.

Reply:
``
[ERROR]   KafkaInstanceConverterTest>KafkaConverterTestBase.testKafkaAffinityToV1Beta1:219->KafkaConverterTestBase.convertKafkaAffinityToV1beta1:236 » IllegalArgument
[ERROR]   KafkaInstanceConverterTest>KafkaConverterTestBase.testZookeeperAffinityToV1Beta1:245->KafkaConverterTestBase.convertKafkaAffinityToV1beta1:236 » IllegalArgument
[ERROR]   KafkaJsonNodeConverterTest>KafkaConverterTestBase.testKafkaAffinityToV1Beta1:219->KafkaConverterTestBase.convertKafkaAffinityToV1beta1:236 » IllegalArgument
[ERROR]   KafkaJsonNodeConverterTest>KafkaConverterTestBase.testZookeeperAffinityToV1Beta1:245->KafkaConverterTestBase.convertKafkaAffinityToV1beta1:236 » IllegalArgument
``

Reply:
This breaks reverse()

Reply:
OK, then lets get rid of it. Probably came from before I implemented `reverse()`.